Dockerfile

```diff
@@ -1,20 +1,39 @@
 # Keep this in sync with the Node version specified for the Conda dev environment (in environment_dev.yml)
-FROM node:22-alpine
+FROM node:22-slim AS build
 
 RUN corepack enable
 
 WORKDIR /usr/src/app
 
-COPY . .
+COPY . ./
+
 # Use the --immutable flag as a precaution. This guarantees that the Docker
 # build will fail if `yarn install` results in a different yarn.lock file than
 # the one checked into the repo. This could happen for example if the version of
 # Yarn used in one dev environment differs from the version used here.
 RUN --mount=type=cache,target=/root/.yarn YARN_CACHE_FOLDER=/root/.yarn \
-    yarn install --immutable --network-timeout 600000 && \
-    if [[ ! -f ".env" ]]; then mv .env.example .env; fi;
+    yarn install --immutable && \
+    yarn run build && \
+    yarn run webpack:prod bundle
+
+# ---------------------------------------------------------------------------------
+# for production
+FROM node:22-slim AS prod
+
+WORKDIR /usr/src/conda-store-ui
+
+COPY --from=build /usr/src/app/app /usr/src/conda-store-ui/app
+COPY --from=build /usr/src/app/dist /usr/src/conda-store-ui/dist
+
+RUN cd app && npm install
+
+EXPOSE 8000
 
+CMD ["node", "app/app.js"]
 
+# ---------------------------------------------------------------------------------
+# for dev
+FROM build AS dev  
```

Comment:
I'm not quite sure I understand this, can you help me understand what this does, how it is intended to be used?

Reply:
This is used to build a dev image. This change introduces 3 build stages.

The first (build) stage (`FROM node:22-slim AS build`) will copy the project into the image and install and build everything (including the bundle for prod).

There is a prod target `FROM node:22-slim AS prod`. This will copy just the built bundle + the webserver into a new image and install all the dependencies for running the webserver.

The dev target `FROM build AS dev ` uses all the same stuff as the build stage and defines the default command to run as `yarn run start:ui`. So if you just run the image without any other arguments it will run the application. This one is used by [docker-compose.yml](https://github.com/conda-incubator/conda-store-ui/blob/main/docker-compose.yml#L82).

Reply:
Makes sense, thanks for the explanation! I see now too that you added the dev target in the to the compose file in this PR.